contracts/crypto-verify/src/contract.rs

```diff
@@ -62,7 +73,41 @@ pub fn query_verify_cosmos(
     let hash = Sha256::digest(message);
 
     // Verification
-    let result = deps.api.secp256k1_verify(&*hash, signature, public_key);
+    let result = deps
+        .api
+        .secp256k1_verify(hash.as_ref(), signature, public_key);
+    match result {
+        Ok(verifies) => Ok(VerifyResponse { verifies }),
+        Err(err) => Err(err.into()),
+    }
+}
+
+pub fn query_verify_ethereum(
+    deps: Deps,
+    message: &[u8],
+    signature: &[u8],
+    signer_address: &str,
+) -> StdResult<VerifyResponse> {
+    // Hashing
+    let mut hasher = Keccak256::new();
+    hasher.update(format!("\x19Ethereum Signed Message:\n{}", message.len()));
+    hasher.update(message);
+    let hash = hasher.finalize();
+
+    // Decompose signature
+    let (v, rs) = match signature.split_last() {
```

Comment:
The term "signature" is a bit ambigues at this low level. An Ethereum signature often means the triple (r, s, v). In cosmos only r and s are in the signature because it does not use pubkey recovery.